File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"

bool is_armstrong_number(int candidate) {
	
	// Check whether the candidate is less than zero
	if (candidate < 0)
		return false;

	int number_of_digits = 0;

	// Get the number of digits in candidate
	for (int tmp_candidate = candidate; tmp_candidate !=0;) {
		tmp_candidate /= 10;
		number_of_digits++;
	}

	int armstrong_result = 0;

	// Loop through all digits and raise them with the number of digits
	for(int tmp_candidate = candidate, digit; tmp_candidate != 0;) {
		digit = tmp_candidate % 10;
		armstrong_result += pow(digit, number_of_digits);
		tmp_candidate /= 10;
	}

	// Return whether the armstrong_result equals candidate
	return (armstrong_result == candidate);
}
	

	

File: python/armstrong-numbers/armstrong_numbers.py
def is_armstrong_number(armstrong_candidate: int) -> bool:

    if armstrong_candidate < 0:
        return False

    armstrong_candidate_s = str(armstrong_candidate)
    armstrong_sum = 0

    for digit_s in armstrong_candidate_s:
        armstrong_sum += int(digit_s) ** len(armstrong_candidate_s)

    return True if armstrong_sum == armstrong_candidate else False

